lib/open_food_network/reports/line_items.rb

```diff
@@ -12,9 +12,7 @@ def orders
       end
 
       def list(line_item_includes = nil)
-        line_items = @order_permissions.
-          visible_line_items.
-          merge(Spree::LineItem.where(order_id: orders.result))
+        line_items = @order_permissions.visible_line_items.in_orders(orders.result)
```

Comment:
I think this is not correct, I cant remember why. I think it's because of the SQL that visible_line_items generates, you cannot simply add a where clause, you need to use merge, which does something else, I cant remember what...

Reply:
I am sure I have tried this before and it didnt work. I cant remember why....

Reply:
I know there can be issues with multiple #where calls directly in the same query chain, but I think its different if they're in named scopes?

Reply:
I didnt mean there would be a problem on the rails side building the query. I meant that I think the query that results from this is not correct for what we need.

Reply:
The #merge here gives us the intersection of both queries, right? So we could write it like this:
```ruby
line_items_in_orders_result = Spree::LineItem.where(order_id: orders.result)
visible_line_items = @order_permissions.visible_line_items
visible_line_items_in_orders_result = visible_line_items.merge(line_items_in_orders_result)
```

Is that not equivalent to `@order_permissions.visible_line_items.in_orders(orders.result)`?

Reply:
I called #to_sql on both versions and the output matches :+1: